### ISSUE TITLE
PIXI v4 + Cocoon : createTouchHook

### ISSUE THREAD

**Issue by JiDW** [pixijs/pixi.js#2648]
Hi there!

New version, new incompatibilities issues.

Let's start right away : what is this function ? 

```javascript
AccessibilityManager.prototype.createTouchHook = function()
{
    var hookDiv = document.createElement('button');
    hookDiv.style.width = 1 + 'px';
    hookDiv.style.height = 1 + 'px';
    hookDiv.style.position = 'absolute';
    hookDiv.style.top = -1000+'px';
    hookDiv.style.left = -1000+'px';
    hookDiv.style.zIndex = 2;
    hookDiv.style.backgroundColor = '#FF0000';
    hookDiv.title = 'HOOK DIV';

    hookDiv.addEventListener('focus', function(){

        this.isMobileAccessabillity = true;
        this.activate();
        document.body.removeChild(hookDiv);

    }.bind(this));

    document.body.appendChild(hookDiv);

};
```

It create an element out of the air and put it on top of PIXI. For what and why ?
I think PIXI should stay within the boundaries of its view (the canvas element) or at least provide some fallbacks when needed. If not, it can't be used anymore as a rendering engine outside of a standard webview. Here for example, Cocoon can't use the DOM so this function break the javascript.

**Comment by photonstorm** [comment]
It's for creating accessible games, something extremely important on desktop. On mobile things are different though, and the DOM approach isn't really suitable. Allowing Sprites to be accessible is well within the remit of Pixi imho, although this should be easily turned off. Have you checked to see if any globals exist to do just that?

**Comment by up209d** [comment]
@JiDW It was there long time ago, have you read the article on GoodBoyDigital Blog ?
http://www.goodboydigital.com/pixi-becomes-accessible/

**Comment by ivanpopelyshev** [comment]
@JiDW I agree that there must be some kind of switch in PIXI to not create DOM elements without prompt.

Offtopic, i didnt move pixi-compressed-textures to v4 yet. Do you remember? That plugin has your code inside.

**Comment by JiDW** [comment]
@photonstorm Ok thank you! I didn't find any globals to turn it off.

@up209d Thank you, no I didn't. We freezed our PIXI version for our last game before this feature (or at least implemented like that) and we're switching to v4 for our next project, that's when I saw this incompatibility.

@ivanpopelyshev We are building our own framework on top of PIXI (based on our previous project) and we used middlewares to deal with the compressed texture and other features. I'll see with my boss what we can release in OpSrc. I think some of them could be a welcome addition to PIXI.

If the Accessibility Manager is only useful for screen reader and TAB key, I think it could be opt-in, or at least only initialized at the first "accessible" DisplayObject created ?

**Comment by ivanpopelyshev** [comment]
@JiDW I will be happy to help you with new project, especially because I have my own 2.5d fork: https://gameofbombs.github.io/pixi-bin/index.html?s=flip&f=cards.js&title=Cards

I'll be happy to provide you features that you need with new plugins and not just freezing the stuff. I'm already working with RPGMaker MV and few other projects on that basis.

I think PIXI v4 will be released this week.

**Comment by GoodBoyDigital** [comment]
Good shout, agree with  @photonstorm - lets look to add an option to disable the accessibility if required. v4 accessibility is should be compatible with mobile devices now. I know for sure it works with Safari 👍

**Comment by englercj** [comment]
Fixed by #2946
